app/Actions/TwoFactorAuth/GenerateQrCodeAndSecretKey.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Actions\TwoFactorAuth;
+
+use BaconQrCode\Renderer\Image\ImagickImageBackEnd;
+use BaconQrCode\Renderer\ImageRenderer;
+use BaconQrCode\Renderer\RendererStyle\RendererStyle;
+use BaconQrCode\Writer;
+use App\Models\User;
+use PragmaRX\Google2FA\Google2FA;
+
+class GenerateQrCodeAndSecretKey
+{
+    public string $companyName;
+
+    /**
+     * Generate new recovery codes for the user.
+     *
+     * @return array{string, string}
+     */
+    public function __invoke($user): array
+    {
+
+        $google2fa = new Google2FA;
+        $secret_key = $google2fa->generateSecretKey();
+
+        $this->companyName = 'Auth';
+        if (is_string(config('app.name'))) {
+            $this->companyName = config('app.name');
+        }
+
+        $g2faUrl = $google2fa->getQRCodeUrl(
+            $this->companyName,
+            (string) $user->email,
+            $secret_key
+        );
+
+        $writer = new Writer(
+            new ImageRenderer(
+                new RendererStyle(800),
+                new ImagickImageBackEnd
```

Comment:
Not sure if the Imagick backend should be used. Because it is not a requirement for laravel itself.

Maybe use SvgImageBackEnd or the GDLib Renderer?
https://github.com/Bacon/BaconQrCode/blob/main/README.md#available-image-renderer-back-ends

Reply:
For sure! Thanks for the quick comment 👍 

That's a good call. I'll try out the SVGImageBackEnd and see how that works.

Reply:
Thanks @Jubeki 👏

I've tested and implemented using the `SvgImageBackEnd()`, which is better than using the Imagick base64 PNG. Appreciate the input!